### ISSUE TITLE
Lifetime errors with skipped `&'static` variable

### ISSUE THREAD

**Issue by epage** [serde-rs/serde#1078]
I needed to evolve my format, so I went from (simplified)
```rust
#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields)]
pub struct SassOptions {
    #[serde(skip)]
    pub import_dir: &'static str,
    pub style: SassOutputStyle,
}

impl Default for SassOptions {
    fn default() -> SassOptions {
        ...
    }
}

#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields, default)]
pub struct ConfigBuilder {
    #[serde(skip)]
    pub root: path::PathBuf,
    pub source: String,
    pub destination: String,
    pub sass: sass::SassOptions,
}
```
to
```rust

```rust
#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields)]
pub struct SassOptions {
    #[serde(skip)]
    pub import_dir: &'static str,
    pub style: SassOutputStyle,
}

impl Default for SassOptions {
    fn default() -> SassOptions {
        ...
    }
}

#[derive(Debug, PartialEq, Default)] 
#[derive(Serialize, Deserialize)]    
#[serde(deny_unknown_fields)]        
pub struct AssetsBuilder {           
    pub sass: sass::SassOptions,     
}                                    

#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields, default)]
pub struct ConfigBuilder {
    #[serde(skip)]
    pub root: path::PathBuf,
    pub source: String,
    pub destination: String,
    pub assets: AssetsBuilder,
}
```

and I started getting lifetime errors
```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'de` due to conflicting requirements
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime 'de as defined on the impl at 95:21...
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
note: ...so that types are compatible (expected std::convert::From<<__A as serde::de::MapAccess<'_>>::Error>, found std::convert::From<<__A as serde::de::MapAccess<'de>>::Error>)
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that types are compatible (expected serde::Deserialize<'_>, found serde::Deserialize<'static>)
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate

error: aborting due to 2 previous errors

error: Could not compile `cobalt-bin`.
```

Situations
- Adding a layer caused the warning
- Switching `import_dir` to `String` removed the error
- Adding a member to `AssetsBuilder` did not fix things.

Original source: https://github.com/cobalt-org/cobalt.rs
Broken source: not uploaded at the moment

**Comment by epage** [comment]
Broken source: https://github.com/epage/cobalt.rs/tree/serde-broken

**Comment by dtolnay** [comment]
Thanks for the detailed report! This is a bug. I have a fix in #1079 and will try to get a release out in a few minutes.

**Comment by dtolnay** [comment]
I released serde_derive 1.0.18 with a fix and confirmed that your code compiles.

**Comment by epage** [comment]
Thanks!
